client/sites/plan/components/current-plan-panel/index.tsx

```diff
@@ -57,14 +59,25 @@ export default function CurrentPlanPanel() {
 		return <PlanPricing inline />;
 	};
 
+	const [ isManageAddOnsModalOpen, setIsManageAddOnsModalOpen ] = useState( false );
+	const onManageAddOnsButtonClick = () => {
+		setIsManageAddOnsModalOpen( true );
+	};
+
 	const renderManageAddOnsButton = () => {
 		if ( isA4APlan || is100YearPlan ) {
 			return null;
 		}
 		return (
-			<Button variant="tertiary" href={ `/add-ons/${ site?.slug }` }>
-				{ translate( 'Manage add-ons' ) }
-			</Button>
+			<>
+				<Button variant="tertiary" onClick={ onManageAddOnsButtonClick }>
+					{ translate( 'Manage add-ons' ) }
+				</Button>
```

Comment:
[nit] This seems to be a bit duplicate. How about creating a `ManageAddOn` component that displays the button and manages the modal internally to avoid repetition?

Reply:
Oh, good idea. I am actually planning to add Track events later; I'll do that in that PR instead!